tools/mason/MasonSearch.chpl

```diff
@@ -133,16 +137,55 @@ proc masonSearch(ref args: list(string)) {
    are displayed first */
 proc rankResults(results: list(string), query: string): [] string {
   use Sort;
+  var res = rankOnScores(results);
   record Comparator { }
   proc Comparator.compare(a, b) {
     if a.toLower().startsWith(query) && !b.toLower().startsWith(query) then return -1;
     else if !a.toLower().startsWith(query) && b.toLower().startsWith(query) then return 1;
     else return 1;
   }
   var cmp : Comparator;
+  if query != ".*" then sort(res, comparator=cmp);
+  return res;
+}
+
+/* Returns a map of packages found in cache along with their scores */
+proc getPackageScores(res: list(string)) {
+  use Map;
+  const pathToReg = MASON_HOME + "/mason-registry/cache.toml";
+  var cacheExists: bool = false;
+  if isFile(pathToReg) then cacheExists = true;
+  const parse = open(pathToReg, iomode.r);
+  const cacheFile = owned.create(parseToml(parse));
+  var packageScores: map(string, int);
+  var packageName: string;
+  const defaultScore = 5;
```

Comment:
No change needed, but I'll note that when we move to a normalized score (0-100, e.g.), I think we want the default to be the max.